src/util/thpool/thpool.c

```diff
@@ -56,11 +58,12 @@ typedef struct job {
 
 /* Job queue */
 typedef struct jobqueue {
-	pthread_mutex_t rwmutex; /* used for queue r/w access */
-	job *front;              /* pointer to front of queue */
-	job *rear;               /* pointer to rear  of queue */
-	bsem *has_jobs;          /* flag as binary semaphore  */
-	int len;                 /* number of jobs in queue   */
+	pthread_mutex_t rwmutex; 		/* used for queue r/w access */
+	job *front;              		/* pointer to front of queue */
+	job *rear;               		/* pointer to rear  of queue */
+	bsem *has_jobs;          		/* flag as binary semaphore  */
+	int len;                 		/* number of jobs in queue   */
+	uint64_t max_queued_queries;		/* limit on number of queries */
```

Comment:
rename from `max_queued_queries` to `cap` this is a generic thread pool (not necessarily running queries)